### ISSUE TITLE
Check usage of unit system conversions

### ISSUE THREAD

**Issue by kjacobsen16** [USGS-WiM/NSSServices#222]
In StreamStats, it specifies the US customary unit system in the NSS scenario calls using a url parameter ("unitsystem=2"). In cases where there are mixed units in the regression equations (like in the PR Flow Duration Statistics), this creates issues. In the case of PR, one of the parameters in the equation (ETPENMON) is in millimeters, but the other parameters are in US customary. In the response from the services, the parameter value and unit comes back as millimeters, but the max/min limits have been converted to inches so it looks like the value is outside of the limits when it isn't (side note, but those aren't rounded either).
![image](https://user-images.githubusercontent.com/35497731/124035496-3b7f1880-d9c2-11eb-8386-925d52e09118.png)

We need to investigate further where this conversion is happening and what will be affected if we decide to remove that unitsystem conversion from the calls. Since Pete is expecting more reports to come in with mixed units, it will be better to fix this issue long term than to ask authors to revise equations to have all one unit system.

Basically, we need to find out how impactful removing that unit system url parameter would be.

In the NSSAgent, I see a "getUnitConversionFactor" function - this is the place to start investigating this

**Comment by kjacobsen16** [comment]
Email sent off to Pete - turns out the unit system conversion is only currently working for the min/max limit of the basin characteristics and the unit and final value of the equation. Since all of the current equations are in cubic feet per second, the latter doesn't do anything.

Note that the unit system conversion is being called in several other places, but there are two issues:
1. In most places where the "getUnitConversionFactor()" function is being called, the "to" unit system is just sent as the unit system of the current unit, not the one designated in the url, so it's not doing anything. https://github.com/USGS-WiM/NSSServices/blob/dev/NSSAgent/NSSAgent.cs#L687
2. In places where the "getUnit()" function is called, a new unit is being created, which doesn't include the "UnitConversionFactorsIn" and other conversion factor objects, resulting in an error in the getUnit function. https://github.com/USGS-WiM/NSSServices/blob/dev/NSSAgent/NSSAgent.cs#L620

The plan I proposed to Pete is to remove the unit system url parameter from StreamStats and NSS and later remove it completely from the code.

**Comment by kjacobsen16** [comment]
Completed, issues added:
https://github.com/USGS-WiM/StreamStats/issues/1124
https://github.com/USGS-WiM/NSS/issues/482
https://github.com/USGS-WiM/NSSServices/issues/223
